docs/Roles.md

```diff
@@ -1,96 +1,39 @@
+The FHIR server uses a role-based access control system. The access control model is based on the following concepts:
 
-```json
-[
-    {
-        "name": "clinician",
-        "resourcePermissions": [
-            {
-                "actions": [
-                    "Read",
-                    "Write"
-                ]
-            }
-        ]
-    },
-    {
-        "name": "researcher",
-        "resourcePermissions": [
-            {
-                "actions": [
-                    "Read"
-                ]
-            }
-        ]
-    },
-    {
-        "name": "admin",
-        "resourcePermissions": [
-            {
-                "actions": [
-                    "Read",
-                    "Write",
-                    "HardDelete"
-                ]
-            }
-        ]
-    }
-]
-```
-
-This structure is passed to the FHIR server at startup and enforced using the `roles` claim in the JWT access token presented when consuming the FHIR server API. 
+- **Data Actions** refer to specific allowed or disallowed operations that can be performed on a FHIR server's data. Examples include `read`, `write`, and `delete`.
+- **Role definitions** or simply **roles**, are named collections of actions that are allowed be performed. They apply to a set of **scopes**.
+- **Scopes** define the subset of data to which a role definition applies. Currently, only the root scope (`/`) is supported, which means that role definitions apply to all the data in the FHIR server.
+- **Role assignments** grants a role definition to an identity (user, group, or service principal).
 
-When deploying the FHIR server into Azure using the provided [resource manager template](../samples/templates/default-azuredeploy.json) the array of roles can be passed in the `additionalFhirServerConfigProperties` parameter, which will add the roles to the [App Settings](https://docs.microsoft.com/en-us/azure/app-service/web-sites-configure) of the front end [Web App](https://azure.microsoft.com/en-us/services/app-service/web/) running the server. 
+The set of data actions that can be part of a role definition are:
 
-In the app settings the nested array structure must be flattened and added to the `FhirServer:Security:Authorization:Roles` section of the configuration. Specifically, a roles array like:
+- `*` allows all data actions
+- `read` is required for reading and searching resources.
+- `write`is required for creating or updating resources.
+- `delete` is required for deleting resources. Hard-deleting requires this in addition to `hardDelete`
+- `hardDelete` is required, in addition to `delete`, for hard-deleting data.
+- `export` is required for exporting data, in addition to `read`
+- `resourceValidate` is required for invoking the [validate resource operation](https://www.hl7.org/fhir/operation-resource-validate.html).
```

Comment:
Will it be possible to read (or at least confirm the existence of) specific data on server as part of validate? If yes, would it be better for 'read' to include 'resourceValidate'?

Reply:
If validation performs a requires a read (e.g. for referential integrity checking), it will require read permission.